source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst

```diff
@@ -1259,6 +1308,10 @@ implementation::
         topologyVersion = error.topologyVersion
 
         with client.lock:
+            # Don't perform any changes if type=LoadBalanced
```

Comment:
I don't think this is correct for a couple of reasons:

1. For post-handshake state change errors, we only want to skip marking the server Unknown. We still want to clear the connection pool using the serviceId of the errored connection.
2. For pre-handshake errors, we want to clear the connection pool if the error occurred after the initial hello/legacy hello command completed.

Reply:
Changed logic to simply not mark server unknown instead inside the next block.